routers/repo/http.go

```diff
@@ -282,6 +282,10 @@ func HTTP(ctx *context.Context) {
 	}
 
 	if !repoExist {
```

Comment:
Can we check to only allow POST method ?
```suggestion
	if !repoExist && ctx.Req.Method == "POST" {
```

Reply:
No, because first call to refs is a GET before a POST, and if it fails then git stops.

Reply:
Can we lie to the first response and not create repo at the first request ?

Reply:
Looking at https://git-scm.com/book/en/v2/Git-Internals-Transfer-Protocols#_the_smart_protocol
We should send an empty response (what an empty repo should look like) at GET and really create the repo at the post request POST http://server/simplegit-progit.git/git-receive-pack

Reply:
Okay, I'll look at it in a bit

Reply:
From what I have tested (`git-receive-pack .` in a bare repo) we should return something like : 
```
001f# service=git-receive-pack
008d0000000000000000000000000000000000000000 capabilities^{}report-status delete-refs side-band-64k quiet atomic ofs-delta agent=git/2.25.0
0000
```

Reply:
I cannot seem to get this to play nice with git. Sending a dummy payload still seems to make the client error, though I can't figure out why.  
I've more or less copied this code, swapping out `refs` with the dummy payload
```
008d0000000000000000000000000000000000000000 capabilities^{}report-status delete-refs side-band-64k quiet atomic ofs-delta agent=git/2.25.0
0000
```
 https://github.com/go-gitea/gitea/blob/d32eac44b16c7b3320b586f402621de84fafa185/routers/repo/http.go#L542-L560  
And still the client is unhappy. I'm not sure if the command is keeping the connection alive longer somehow, but the client complains of Gitea hanging up too quickly.

Reply:
Ok, I haven't all the details but I was totally wrong the response should be a base64 of 
```
001e# service=git-upload-pack
00000000
```
I captured the response of https://github.com/sapk-fork/empty-repo.git/info/refs?service=git-upload-pack (need to be logged of since github seems to filter on that)

Reply:
Oh I just use upload and not receive ... I will re-have a look later.

Reply:
I figured it out, apparently there was a space that was getting lost
`capabilities^{}report-status `
`capabilities^{} report-status `

Reply:
It's not a space, it's actually a NUL byte. `\000`  
TIL